src/reducers/setCursor.ts

```diff
@@ -38,18 +38,20 @@ const setCursor = (
     path: Path | null
   },
 ): State => {
-  if (path && path.length > 1 && equalThoughtRanked(path[0], HOME_PATH[0])) {
+  if (path && path.length > 1 && path[0] === HOME_TOKEN) {
     // log error instead of throwing since it can cause the pullQueue to enter an infinite loop
     console.error(
       new Error(
-        `setCursor: Invalid Path ${JSON.stringify(pathToContext(path))}. Non-root Paths should omit ${HOME_TOKEN}`,
+        `setCursor: Invalid Path ${JSON.stringify(
+          pathToContext(state, path),
+        )}. Non-root Paths should omit ${HOME_TOKEN}`,
       ),
     )
     return state
   }
 
-  const simplePath = path ? simplifyPath(state, path) : HOME_PATH
-  const context = pathToContext(simplePath)
+  const simplePath = (path || HOME_PATH) as SimplePath
```

Comment:
I forgot to remove context view-related logic here. In many places, I have done that. We need to discuss how to fix the context view.  For now, I think it will be okay to remove this context view logic here. What do you think?

Reply:
Sort of. The distinction between `Path` and `SimplePath` is related to context view, but it has more to do with the type system than anything. `Paths` can include cycles and `SimplePaths` cannot. Coercing a `Path` to a `SimplePath` before it has been simplified is very dangerous because we are making the type system lie to us. We rely on the type system to show us where there are errors and to help us with refactoring. That's why I consider this a more sensitive and risky change than simply disabling some functionality.

If `simplifyPath` is not working yet, we can have `simplifyPath` do the type coercion and include a MIGRATION_TODO there. Then we only have to fix it one place. Silently disabling `simplifyPath` in an unknown number of call sites is going to be a pain to track down.

Factoring out `simplifyPath` completely via a re-architecture is something we may do down the line, but we want to do that on a stable base with context view working so that we have context view unit tests to back up the refactor 🙂.

Reply:
> Sort of. The distinction between `Path` and `SimplePath` is related to context view, but it has more to do with the type system than anything. `Paths` can include cycles and `SimplePaths` cannot. Coercing a `Path` to a `SimplePath` before it has been simplified is very dangerous because we are making the type system lie to us. We rely on the type system to show us where there are errors and to help us with refactoring. That's why I consider this a more sensitive and risky change than simply disabling some functionality.
> 
> If `simplifyPath` is not working yet, we can have `simplifyPath` do the type coercion and include a MIGRATION_TODO there. Then we only have to fix it one place. Silently disabling `simplifyPath` in an unknown number of call sites is going to be a pain to track down.

Understood. `simplifyPath` seems to be working. I think it will be better to keep `simplifyPath` for now. We are going to fix the context view later.